rasa/core/brokers/pika.py

```diff
@@ -147,7 +147,10 @@ async def from_endpoint_config(
     async def connect(self) -> None:
         """Connects to RabbitMQ."""
         self._connection = await self._connect()
-        self._connection.add_reconnect_callback(self._publish_unpublished_messages)
+        if self._connection is None:
+            return None
```

Comment:
`self._connection is None` can never happen here, since `self._connect()` returns a non-optional. Was this added to make the linter happy? If so, I think it's better to make the linter ignore the false positive here.

Reply:
Yep, I only added this because of a `mypy` `union-attr complaint and I think that's because `self._connection` is initialised as `Optional`. I'll ignore this in that case.